packages/api-client-core/src/GadgetConnection.ts

```diff
@@ -114,13 +115,21 @@ export class GadgetConnection {
 
   enableSessionMode(options?: true | BrowserSessionAuthenticationModeOptions) {
     this.authenticationMode = AuthenticationMode.BrowserSession;
+
+    let sessionTokenStore;
     if (!options || typeof options == "boolean" || options.storageType == BrowserSessionStorageType.Durable) {
-      this.sessionTokenStore = window.localStorage;
+      sessionTokenStore = window.localStorage;
     } else if (options.storageType == BrowserSessionStorageType.Session) {
-      this.sessionTokenStore = window.sessionStorage;
+      sessionTokenStore = window.sessionStorage;
     } else {
-      this.sessionTokenStore = new InMemoryStorage();
+      sessionTokenStore = new InMemoryStorage();
+    }
+
+    if (options && isObject(options) && "initialToken" in options && options.initialToken) {
```

Comment:
nit: I don't think the check of `options` is necessary here. I believe `isObject` returns false if `options` is undefined

Reply:
Oh yep, thanks!